### PR TITLE
add role, rolebinding and servicemonitor for metrics

### PR DESCRIPTION
**Pull request by rakeshgm**
give permissions for prometheus-k8s
service account to scrape metrics.

### CODE REVIEW

**Comment by rakeshgm** [comment]
This PR is only on OCP dir and not k8s. I will send that in the next PR

**Review by ShyamsundarR — COMMENTED** [review]
Still reviewing and checking bundles to ensure these are packaged.

File: config/rbac/metrics_role.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+kind: Role
```
Comment (ShyamsundarR) [inline config/rbac/metrics_role.yaml:1]: (question/doubt) I still wonder why we cannot use https://github.com/RamenDR/ramen/blob/main/config/rbac/auth_proxy_client_clusterrole.yaml instead.

Comment (ShyamsundarR) [reply config/rbac/metrics_role.yaml:1]: IOW, if we do not need the above file we should remove it?

Comment (rakeshgm) [reply config/rbac/metrics_role.yaml:1]: there are errors in prometheus-k8s pods when there is are no permissions. 
```
* failed to list *v1.Service: services is forbidden
* failed to list *v1.Pod: pods is forbidden
* failed to list *v1.Endpoints: endpoints is forbidden
```

File: config/hub/default/ocp/metrics_role_binding.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ramen-metrics
```
[the remark this answers is not in the document]
Comment (rakeshgm) [reply config/hub/default/ocp/metrics_role_binding.yaml]: updated

File: config/rbac/metrics_role.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: ramen-metrics
```
[the remark this answers is not in the document]
Comment (rakeshgm) [reply config/rbac/metrics_role.yaml]: updated

**Review by ShyamsundarR — APPROVED** [review]

**Pull request merged**